plumbing/protocol/packp/updreq_decode.go

```diff
@@ -83,14 +83,16 @@ func (req *ReferenceUpdateRequest) Decode(r io.Reader) error {
 		rc = io.NopCloser(r)
 	}
 
-	d := &updReqDecoder{r: rc, s: pktline.NewScanner(r)}
+	d := &updReqDecoder{r: rc, s: r}
 	return d.Decode(req)
 }
 
 type updReqDecoder struct {
 	r   io.ReadCloser
-	s   *pktline.Scanner
+	s   io.Reader
```

Comment:
The field `s` sounds confusing now that this is no longer a scanner.

```suggestion
	pr   io.Reader
```